src/s1reader/s1_orbit.py

```diff
@@ -1,12 +1,169 @@
+import cgi
 import datetime
 import glob
 import os
+import requests
 import warnings
 
-
 # date format used in file names
 FMT = "%Y%m%dT%H%M%S"
 
+# Required for orbit download
+scihub_url = 'https://scihub.copernicus.eu/gnss/odata/v1/Products'
+# Namespaces of the XML file returned by the S1 query. Will they change it?
+w3_url = '{http://www.w3.org/2005/Atom}'
+m_url = '{http://schemas.microsoft.com/ado/2007/08/dataservices/metadata}'
+d_url = '{http://schemas.microsoft.com/ado/2007/08/dataservices}'
+# Scihub guest credential
+scihub_user = 'gnssguest'
+scihub_password = 'gnssguest'
+
+
+def check_internet_connection():
+    '''
+    Check connection availability
+    '''
+    url = "http://google.com"
+    try:
+        requests.get(url, timeout=10)
+    except (requests.ConnectionError, requests.Timeout) as exception:
+        raise sys.exit(exception)
+
+
+def parse_safe_filename(safe_filename):
+    '''
+    Extract info from S1-A/B SAFE filename
+    SAFE filename structure: S1A_IW_SLC__1SDV_20150224T114043_20150224T114111_004764_005E86_AD02.SAFE
+    Parameters:
+    -----------
+    safe_filename: string
+       Path to S1-A/B SAFE file
+    Returns:
+    -------
+    List of [sensor_id, mode_id, start_datetime,
+                end_datetime, abs_orbit_num]
+       sensor_id: sensor identifier (S1A or S1B)
+       mode_id: mode/beam (e.g. IW)
+       start_datetime: acquisition start datetime
+       stop_datetime: acquisition stop datetime
+       abs_orbit_num: absolute orbit number
+    '''
+
+    safe_name = os.path.basename(safe_filename)
+    sensor_id = safe_name[2]
+    sensor_mode = safe_name[4:10]
+    start_datetime = datetime.datetime.strptime(safe_name[17:32],
+                                                '%Y%m%dT%H%M%S')
+    end_datetime = datetime.datetime.strptime(safe_name[33:48],
+                                              '%Y%m%dT%H%M%S')
```

Comment:
```suggestion
    start_datetime = datetime.datetime.strptime(safe_name[17:32], FMT)
    end_datetime = datetime.datetime.strptime(safe_name[33:48], FMT)
```
`FMT` is global used by functions in this module.